Source/WebCore/dom/FragmentDirectiveRangeFinder.cpp

```diff
@@ -99,8 +99,11 @@ static const Node& nearestBlockAncestor(const Node& node)
 {
     const Node* currentNode = &node;
     while (currentNode) {
-        if (!currentNode->isTextNode() && currentNode->renderer() && currentNode->renderer()->style().isDisplayBlockLevel())
-            return *currentNode;
+        if (!currentNode->isTextNode()) {
+            CheckedPtr renderer = currentNode->renderer();
```

Comment:
Calling Node::renderer() doesn't look particularly expensive and it is inline. The commit log says there is a branch in it but all I see is a bit masking operation in `CompactPointerTuple::pointer()`. That said, I don't mind caching it in a local variable. But then you decided to cache in a CheckedPtr, which is definitely more expensive as it does ref-counting. It is also not clear to me why we need a CheckedPtr, it seems we're only calling trivial inline functions on the renderer.

Reply:
The comment in Node.h you're pointing to dates from 2012 and is very outdated. We should remove it. At the time the comment was added, the function looked like this:
```
    // As renderer() includes a branch you should avoid calling it repeatedly in hot code paths.
    RenderObject* renderer() const { return hasRareData() ? m_data.m_rareData->renderer() : m_data.m_renderer; };
```

There was indeed a branch. But not anymore.

Reply:
OK, thanks for pointing out this comment was outdated.

I added the checked pointer because for some reason I got the idea that `RenderObject::style()` might not always be trivial, but I double checked now and it always resolves to trivial calls, so indeed it's unnecessary to use a smart pointer here, my bad.

If you think it's worth caching the `Node::renderer()` calls in a local variable I can update the MR to just do that (and remove that comment), but if it's not worth it we can just close this one. I'm ambivalent because it makes the code a bit less readable for probably very little gain.

Reply:
@rniwa Do you think it's worth it?

Reply:
We should check it but I'm pretty certain clang can optimize this without us doing anything.

Reply:
I'd say it's not worth it, if clang will optimize this, and also the code is more readable as is. So I think it's OK to close this.